### ISSUE TITLE
Inconsistent Access Permissions When Restoring from Backup

### ISSUE THREAD

**Issue by baksha97** [okta/okta-mobile-swift#198]
### Describe the bug?

I've encountered an edge case where restoring from a backup causes inconsistent access permissions due to differing configurations between token storage and metadata storage in `KeychainTokenStorage.swift`.

The access settings and security settings for tokens and metadata are configured differently, leading to inconsistencies:

1. **Accessibility**:
   - **Token Storage (`item`)**: Uses variable `accessibility` (Line 87).
   - **Metadata Storage (`metadataItem`)**: Fixed `.afterFirstUnlock` (Line 98).

2. **Access Control**:
   - **Token Storage (`item`)**: Includes `accessControl` (Line 89).
   - **Metadata Storage (`metadataItem`)**: No `accessControl` (Line 96).

Additionally, the built-in defaults for the SDK itself are:
```swift
public static var standard: [Security] = [.accessibility(.afterFirstUnlockThisDeviceOnly)]
```
This indicates that the default behavior with the fixed accessibility (`.afterFirstUnlock`) for metadata leads to a default mismatching. 

When querying for all credentials, the list contains an ID that shouldn't be there. This inconsistency causes all querying of logins to fail permanently on a user's device (notice how this `try` error propagation short-circuits the `compactMap` - did we mean to use `try?` instead?). This likely results from the inconsistency between the token and metadata access settings, causing only the metadata to be available during backup restoration.

```swift
 func find(where expression: @escaping (Token.Metadata) -> Bool,
              prompt: String? = nil,
              authenticationContext: TokenAuthenticationContext? = nil) throws -> [Credential]
    {
        try allIDs
            .map(tokenStorage.metadata(for:))
            .filter(expression)
            .compactMap({ metadata in
                try self.with(id: metadata.id, prompt: prompt, authenticationContext: authenticationContext)
            })
    }
```

### What is expected to happen?

Both token storage and metadata storage should have consistent access permissions and security settings, ensuring that all credentials are accessible and querying for logins does not fail.

### What is the actual behavior?

Tokens and metadata have inconsistent access permissions, leading to access issues and IDs appearing in the list when they shouldn't. This causes **all** usage of `Credential.find` to fail permanently on a user's device.

### Reproduction Steps

1. Store a token and its metadata using `KeychainTokenStorage`.
2. Create a backup of the keychain.
3. Restore the backup on a **new device.**
4. Query for all credentials.

Believe it or not, I believe this scenario might be limited to a **brand new device**. I'm having trouble replicating it because of how unique the hardware requirements are. 

Given the extensive use of our apps by a large user base, it is not surprising that this issue has surfaced. This came to our attention because we heavily rely on `Credential.find`, which is particularly unforgiving when handling invalid states. It's also **very** likely there are more edge cases/scenarios that cause this invalid state since there's always a chance of a mismatch. 

### Solution (Suggested):

Ensure both token storage and metadata storage use the same `accessibility` and `accessControl` settings for consistency.

```swift
let metadataItem = Keychain.Item(account: id,
                                 service: KeychainTokenStorage.metadataName,
                                 accessibility: accessibility,  // Use the same accessibility as token
                                 accessGroup: accessGroup,
                                 accessControl: accessControl,  // Include accessControl
                                 synchronizable: accessibility.isSynchronizable,
                                 value: try encoder.encode(metadata))
```

### Next Steps
- What are we doing for users who have **already** restored to a brand new device? They are not able to `find` `Credential`s again, permanently.
  - **Potentially addressed** by #199 

### SDK Version(s)

1.6.1

**Comment by mikenachbaur-okta** [comment]
Thank you for the detailed report. I've created PR #200 which should address the issue.

**Comment by mikenachbaur-okta** [comment]
This issue has been resolved in [release 1.8.1](https://github.com/okta/okta-mobile-swift/releases/tag/1.8.1), so I'll close this issue. If you still experience these problems in the future, please feel free to reopen it.
